client/packages/common/src/ui/forms/JsonForms/components/Array.tsx

```diff
@@ -0,0 +1,189 @@
+            {label}:
+          </Typography>
+        </Box>
+        <Box width={FORM_INPUT_COLUMN_WIDTH} textAlign="right">
+          <IconButton
+            icon={<PlusCircleIcon />}
+            label={t('label.add-another')}
+            color="primary"
+            onClick={addItem(
+              path,
+              uischema.defaultNewItem ?? createNewItem(foundUISchema)
+            )}
+          />
+        </Box>
+      </Box>
+      {data.map((child, index) => {
+        return (
+          <Accordion
+            key={index}
+            defaultExpanded={index === data.length - 1}
+            sx={{
+              mt: '0 !important',
+              mb: index === data.length - 1 ? '20px !important' : 1,
+            }}
+          >
+            <AccordionSummary
+              expandIcon={<ChevronDownIcon />}
+              sx={{
+                '&:hover .array-remove-icon': { visibility: 'visible' },
+                margin: '0 !important',
+                minHeight: '0 !important',
+                '.MuiAccordionSummary-content': {
+                  margin: '5px !important',
+                },
+                '.Mui-expanded': {
+                  marginBottom: '0 !important',
+                },
+              }}
+            >
+              <Box
+                display="flex"
+                width={FORM_LABEL_COLUMN_WIDTH}
+                justifyContent="space-between"
+                alignItems="center"
+              >
+                <IconButton
+                  icon={<MinusCircleIcon />}
+                  label={t('label.remove')}
+                  color="primary"
+                  className="array-remove-icon"
+                  sx={{ visibility: 'hidden' }}
+                  onClick={removeItems(path, [index])}
+                />
+                <Typography
+                  sx={{
+                    fontWeight: 'bold',
+                    textAlign: 'end',
+                    whiteSpace: 'nowrap',
+                  }}
+                >
+                  {uischema?.itemLabel
+                    ? RegexUtils.formatTemplateString(uischema?.itemLabel, {
+                        ...child,
+                        index: index + 1,
+                      })
+                    : index + 1}
+                </Typography>
+              </Box>
+            </AccordionSummary>
+            <AccordionDetails>
+              <JsonFormsDispatch
+                key={index}
+                schema={schema}
+                uischema={foundUISchema}
+                enabled={enabled}
+                path={`${path}.${index}`}
+              />
+            </AccordionDetails>
+          </Accordion>
+        );
+      })}
+    </Box>
+  );
+};
+
+const withJsonFormsArrayCustomProps = (
+  component: ComponentType<ArrayControlCustomProps>
+) =>
+  withJsonFormsArrayControlProps(component as ComponentType<ArrayControlProps>);
+
+export const Array = withJsonFormsArrayCustomProps(UIComponent);
```

Comment:
Seems a bit hacky -- just a wrapper for the base function so it can accept the component with extended props.

I don't really get why Typescript allows the `as` here, but doesn't let me put `UIComponent as ComponentType<ArrayControlProps>` directly in the base function (withJsonFormsArrayControlProps)

Reply:
You mean just:
`export const Array = withJsonFormsArrayControlProps(UIComponent as ComponentType<ArrayControlProps>);`
seems to be ok here, no idea why...